doc/ARS/Local_ARS_HLD.md

```diff
@@ -0,0 +1,892 @@
+    - Per packet port selection
+2. Support enabling ARS over NHG
+3. Support path quality configuration
+4. Support ACL action to disable ARS 
+
+* Phase 2
+    - Support enabling ARS over LAG
+    - Alternative path support
+    - Statistics
+
+
+### Architecture Design 
+
+The following orchestration agents will be added or modified. The flow diagrams are captured in a later section.
+
+![Orchestration agents](images/config_flow.png)
+__Figure 3: Orchestration agents__
+
+#### ArsOrch
+arsorch - it is an orchestration agent that handles the configuration requests from CONFIG_DB. It is responsible for creating the SAI ARS profile and configuring SAI ARS object. 
+
+#### Orchdaemon
+orchdaemon - it is the main orchestration agent, which handles all Redis DB's updates, then calls appropriate orchagent, the new arsorch should be registered inside the orchdaemon.
+
+#### RouteOrch
+routeOrch monitors operations on Route related tables in APPL_DB and converts those operations in SAI commands to manage IPv4/IPv6 route and nexthops. New functionality for nexthop group change notification and configuring ARS-enabled NHG.
+
+#### PortsOrch
+portsorch handles all ports-related configurations. New functionality for enabling ARS on ports.
+
+#### AclOrch
+aclorch is used to deal with configurations of ACL table and ACL rules. New rule action for control ARS operation.
+
+
+ARS is closely tied to vendor-specific hardware implementations, requiring thorough validation of all configurations using SAI capabilities.
+
+### High-Level Design 
+
+- A standard ARS configuration consists of creating an ARS profile and enabling ARS on the designated ports. This process also includes activating ARS for nexthop groups (Adaptive Routing) and LAGs (Adaptive Switching). At present, configurations are carried out manually via CONFIG_DB, with potential future enhancements to incorporate management through an external controller or routing protocol extensions.
+
+
+- New orchagent: ArsOrch responsible for 
+1. Creating/Updating ARS profile 
+2. Enable interfaces for ARS
+3. Enable ARS over nexthop group
+4. Allow configuration of path metrics
+
+- ACL changes<br>
+When a new ACL table is created, SAI needs to receive a list of supported actions which the rules belonging to this table are allowed to use.
+To support the new ARS disable action, the custom table types table schema will be extended with a ARS action attribute - "ARS_ACTION" for the actions attribute field.
+
+The diagrams below illustrate the typical sequences for ARS configuration, showcasing key workflows for both Adaptive Routing and Adaptive Switching.
+
+#### Sequence diagrams
+
+__Figure 4: Initilization flow__
+![](images/init_seq.png)
+
+1. Bind to Events
+The ArsOrch component binds to relevant events to start monitoring and processing ARS-related interfaces.
+
+2. Get ARS SAI Capabilities
+ArsOrch retrieves ARS capabilities from the SAI layer and saves them in STATE_DB.

```

Comment:
Please add a schema for STATE_DB to list what all capabilities are being saved in the DB.